x/ibc-hooks/README.md

```diff
@@ -114,6 +114,40 @@ In wasm hooks, post packet execution:
 * if wasm message has error, return ErrAck
 * otherwise continue through middleware
 
+## Ack callbacks
 
+A contract that, through a wasm hook, sends an IBC transfer, may need to listen for the ACK from that packet. To allow
+contracts to listen on the ack of specific packets, we provide Ack callbacks. 
 
-### Testing strategy
\ No newline at end of file
+### Design
+
+The sender of an IBC transfer packet may specify a callback for when the ack of that packet is received in the memo 
+field of the transfer packet. 
+
+### Use case
+
+The crosschain swaps implementation sends an IBC transfer. If the transfer were to fail, we want to allow the sender
+to be able to retrieve their funds (which would otherwise be stuck in the contract). To do this, we allow users to 
+retrieve the funds after the timeout has passed, but without the ack information, we cannot guarantee that the send 
+hasn't failed (i.e.: returned an error ack notifying that the receiving change didn't accept it)
+
+### Implementation
+
+#### Callback information in memo
+
+The for the callback to be processed, the transfer packet's memo should look like this:
+
+`{"ibc_callback": "osmo1contractAddr"}`
+
+The wasm hooks will keep the mapping from the packet's channel and sequence to the contract in storage. When an ack is
+received, it will notify the specified contract via a sudo message.
```

Comment:
Just realized, contract authors have to handle someone setting a callback they weren't expecting. (So they need to do some validation)

Reply:
Should we add a version field on `ibc_callback` or probably not needed? (e.g. `ibc_callback_v1`)

Reply:
> someone setting a callback the contract wasn't expecting

Senders don't specify the callback format though. Just who will receive it. So as long as the format doesn't change on the chain, contracts should be able to trust the interface. 

I like the idea of versioning when we need it. But since we don't have plans for v2 yet, I think the easiest thing here would be to leave it as is and then add `ibc_callback_v2` if we make a change.